test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java

```diff
@@ -94,4 +136,69 @@ public void writeAndReadBackAMessageWithTlsAndAuthEnabledServer() {
         String readMessage = TestUtils.readNextEventMessage(scopeName, streamName, readerClientConfig, readerGroupName);
         assertEquals(testMessage, readMessage);
     }
+
+    @SneakyThrows
+    @Test
+    public void restApiInvocationWithSecurityEnabled() {
+        String restApiUser = "rest-api-user";
+        String restApiUserPwd = "super-secret";
+
+        final Map<String, String> passwordInputFileEntries = new HashMap<>();
+        passwordInputFileEntries.put("restApiUser", "prn::*,READ_UPDATE");
+
+        // Setup and start the cluster
+        @Cleanup
+        ClusterWrapper cluster = ClusterWrapper.builder()
+                .controllerRestEnabled(true)
+
+                // Auth related configs
+                .authEnabled(true)
+                .passwordAuthHandlerEntries(TestUtils.preparePasswordInputFileEntries(passwordInputFileEntries, restApiUserPwd))
+
+                // TLS related configs
+                .tlsEnabled(true)
+                .tlsServerCertificatePath(TestUtils.pathToConfig() + SecurityConfigDefaults.TLS_SERVER_CERT_FILE_NAME)
+                .tlsServerKeyPath(TestUtils.pathToConfig() + SecurityConfigDefaults.TLS_SERVER_PRIVATE_KEY_FILE_NAME)
+                .tlsHostVerificationEnabled(false)
+                .tlsServerKeystorePath(TestUtils.pathToConfig() + SecurityConfigDefaults.TLS_SERVER_KEYSTORE_NAME)
+                .tlsServerKeystorePasswordPath(TestUtils.pathToConfig() + SecurityConfigDefaults.TLS_PASSWORD_FILE_NAME)
+
+                .build();
+        cluster.start();
+
+        // Setup REST client config
+        org.glassfish.jersey.client.ClientConfig clientConfig = new org.glassfish.jersey.client.ClientConfig();
+        clientConfig.register(JacksonJsonProvider.class);
+        clientConfig.property("sun.net.http.allowRestrictedHeaders", "true");
+        clientConfig.register(HttpAuthenticationFeature.basic(restApiUser, restApiUserPwd));
+
+        // Prepare a TLS context with truststore containing the signing CA's vertificate
+        KeyStore trustStore = createTrustStore(TestUtils.pathToConfig() + SecurityConfigDefaults.TLS_CA_CERT_FILE_NAME);
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        tmf.init(trustStore);
+        SSLContext tlsContext = SSLContext.getInstance("TLS");
```

Comment:
Do any of these need cleaning up?

Reply:
This is alright. It's creating a truststore containing the CA certificate, so that the server certificate can be trusted by the client.

Reply:
Also, this is where we read certs from files, and we close the stream there: 

https://github.com/pravega/pravega/blob/2e70f876c5eaeaf0ed51504174c5e6b13f0645a1/common/src/main/java/io/pravega/common/util/CertificateUtils.java#L40-L47